python/ql/src/semmle/python/Exprs.qll

```diff
@@ -691,6 +691,26 @@ class StrConst extends Str_, ImmutableLiteral {
 
     override Object getLiteralObject() { none() }
 
+    /* Add some utility methods missing on `string` */
+
+    bindingset[s]
```

Comment:
I'm a bit wary about making these require `s` to be bound. As a user of the library, I would expect to be able to call `strconst.startsWith(prefix)` with `prefix` unbound to result in `prefix` being bound to all possible prefixes of `strconst`. 

Of course, I realise that this restriction ultimately comes from `prefix` and `length` which both require their inputs to be bound. What I mean is that there should be a compelling reason to propagate this restriction further, especially if it may result in "surprising" behaviour for the end-user.

Reply:
Further to that point, are there cases where `strconst.startsWith("foo")` cannot be rewritten as `strconst.getText().matches("foo%")`?

Apart from when the string we're matching on contains `_`, `%` or `\`, the behaviour should be the same.